code/script.js

```diff
@@ -1,51 +1,153 @@
+form.addEventListener("submit", (event) => {
+  event.preventDefault();
+  const name = nameInput.value;
+  userAnswer(`${name}`);
+  // This will keep the input box empty after user clicks send
+  nameInput.value = "";
+  setTimeout(() => animalOptions(name), 1000);
+});
+
+// 2 - Bot welcomes you to Hogwarts and asks your favourite animal
+const animalOptions = (name) => {
+  questionStep++;
+  botAnswer(`Welcome ${name} to Hogwarts!`);
+  setTimeout(
+    () =>
+      botAnswer(`In order to sort you into a house you have to answer a few questions...
+  First, what is your favourite animal?`),
+    1000
+  );
+  inputWrapper.innerHTML = `
+    <button id="lionButton">Lion 🦁</button>
+    <button id="snakeButton">Snake 🐍</button>
+    <button id="badgerButton">Badger 🦡</button>
+    <button id="eagleButton">Eagle 🦅</button>
+    `;
+  document.getElementById("lionButton").addEventListener("click", () => {
+    type = "lion";
+    handleInput("Lion!");
+  });
+  document.getElementById("snakeButton").addEventListener("click", () => {
+    type = "snake";
+    handleInput("Snake!");
+  });
+  document.getElementById("badgerButton").addEventListener("click", () => {
+    type = "snake";
+    handleInput("Badger!");
+  });
+  document.getElementById("eagleButton").addEventListener("click", () => {
+    type = "eagle";
+    handleInput("Eagle!");
+  });
+};
 
-// Set up your eventlisteners here
+//3 - Bot asks you to choose favourite colour
+const colourOptions = (type) => {
+  questionStep++;
+  botAnswer(
+    `Interesting! And could you tell me what your favourite colour is?`
+  );
+  inputWrapper.innerHTML = `
+  <select id = 'select'>
+  <option value='' selected disabled> Choose colour </option>
+  <option value='Red'>Red</option>
+  <option value='Green'>Green</option>
+  <option value='Yellow'>Yellow</option>
+  <option value='Blue'>Blue</option>
+  </select> `;
+  const select = document.getElementById("select");
+  select.addEventListener("change", () => handleInput(select.value));
+};
 
-// When website loaded, chatbot asks first question.
-// normally we would invoke a function like this:
-// greeting()
-// But if we want to add a little delay to it, we can wrap it in a setTimeout:
-// setTimeout(functionName, timeToWaitInMilliSeconds)
-// This means the greeting function will be called one second after the website is loaded.
-setTimeout(greetUser, 1000);
+//4 - Bot will place you in Hogwarts house
+const houseStatement = (colour) => {
+  questionStep++;
+  if (colour === "Green") {
+    botAnswer(`SLYTHERIN!`);
+  } else {
+    botAnswer(`Hmm... Tough call... Better put you in...`);
+    setTimeout(() => botAnswer(`SLYTHERIN!`), 2000);
+  }
+
+  //You will love your new house
+  inputWrapper.innerHTML = `
+<button id="yayButton">Yay!</button>
+`;
+  document
+    .getElementById("yayButton")
+    .addEventListener("click", () =>
+      handleInput("I can't wait to become the gratest witch in the world!")
```

Comment:
This doesn't affect the code but there might be a typo in the message on line 141:
 Gratest = Greatest

Reply:
Yes its a typo 😅